planning/behavior_velocity_planner/src/scene_module/detection_area/scene.cpp

```diff
@@ -87,20 +85,21 @@ double calcSignedArcLength(
   const autoware_auto_planning_msgs::msg::PathWithLaneId & path,
   const geometry_msgs::msg::Point & p1, const geometry_msgs::msg::Point & p2)
 {
-  const std::pair<int, double> src = findWayPointAndDistance(path, p1);
-  const std::pair<int, double> dst = findWayPointAndDistance(path, p2);
-  if (dst.first == -1) {
+  const auto src = findWayPointAndDistance(path, p1);
+  const auto dst = findWayPointAndDistance(path, p2);
+  if (!src || !dst) {
     const double dx = p1.x - p2.x;
     const double dy = p1.y - p2.y;
     return -1.0 * std::hypot(dx, dy);
   }
 
-  if (src.first < dst.first) {
-    return calcArcLengthFromWayPoint(path, src.first, dst.first) - src.second + dst.second;
-  } else if (src.first > dst.first) {
-    return -1.0 * (calcArcLengthFromWayPoint(path, dst.first, src.first) - dst.second + src.second);
+  if (src->first < dst->first) {
+    return calcArcLengthFromWayPoint(path, src->first, dst->first) - src->second + dst->second;
```

Comment:
@k-obitsu 
since similar approach is used in other behavior modules , cant't it be replaced with tier4_autoware_utils calcSignedArcLength ?

Reply:
@taikitanaka3 
I will confirm it.

Reply:
This PR should be left as it is to make it the same as iv, but I think they can be replaced. 
This PR adds processing when ego exceeds path, but in [common one](https://github.com/autowarefoundation/autoware.universe/blob/tier4/proposal/common/tier4_autoware_utils/include/tier4_autoware_utils/trajectory/trajectory.hpp#L59-L75), the last point is always found.
I think the difference is whether to [filter in the lateral direction](https://github.com/autowarefoundation/autoware.universe/blob/tier4/proposal/planning/behavior_velocity_planner/src/scene_module/detection_area/scene.cpp#L35).

Reply:
I send PR to use common one. https://github.com/autowarefoundation/autoware.universe/pull/362

Reply:
@k-obitsu 
Can you confirm this change? https://github.com/autowarefoundation/autoware.universe/pull/362
@kosuke55 
merge this PR if @k-obitsu 's approve then merge #362 and write release note please

Reply:
@k-obitsu 
OK, it looks good in code so I approve it. But please fix `build-and-test-differential`.

Reply:
I think this error is realated to https://github.com/autowarefoundation/autoware-github-actions/pull/76. Rebasing may fix it.

Reply:
sorry for late replying, I'll fix it later.

Reply:
Thank you. Rebase may fix it so let's wait for the result.

Reply:
↑ fixed build error